### DISCUSSION TITLE
Rain Rate Retrieval using Power Law

### DISCUSSION THREAD

**Discussion by raffyjef** [ARM-DOE/pyart#1210] [Q&A] [answered]
hi! Does the retrieve.est_rain_rate_z function uses the Z=aR^b relation? I am curious where the default alpha and beta values were derived from? I went to the reference indicated but I did not see how these values were derived. Hope you can answer me. Thank you!

pyart.retrieve.est_rain_rate_z(radar, alpha=0.0376, beta=0.6112, refl_field=None, rr_field=None)

I am asking this because I want to calculate rain rate using the Z-R relation and by changing its coefficient, however the coefficients present doesn't seem to point to coefficients in the actual equation.

**Comment by zssherman** [comment, marked as answer]
Hi @raffyjef I had this discussion with some of the folks who originally wrote the code, posting the discussion below:

```
est_rain_rate_z_poly: that was based on something developed by Zawadski. I am not sure he published something about it.
Maybe you can find a conference paper. Marco may know the exact reference. I got it in a seminar when he visited
MeteoSwiss some time ago. Perhaps you can ask him for the reference? Let me know if he tells you one

The Swiss Z-R is based on long-term distrometer measurements and studies by Juerg Joss, and collaborations
he had with Paul Smith, Gori, Doelling and Zawadzki. You can find a number of references in the
report Joss et al (1998) that Jordi has sent you.

Ideally one would use variable parameters in the Z-R relation, or, even a more sophisticated approach
that does not assume a power law. But to do so, one needs a robust estimation procedure to vary the parameters.
In the absence of such a procedure Juerg decided to use a fixed Z-R relation and we use it still today.
Based on the long-term distrometer data Juerg found that an exponent of 1.5 and a factor around 320 best
fits the Swiss rainfall. As far as I remember the factor was set to 316 because one then gets “nice” numbers
in the dBZ-R lookup table (useful for manual calculations and color legends etc).  I hope this helps.

And in the following paper, 7 years of German data:

Doelling, I. G., J. Joss, and J. Riedl, Systematic variations of Z-R relationships from drop size distributions
measured in Northern Germany during seven years, Atmos. Res., 48, 635–649, 1998
you may see  that
Z =102.5 R1.5 is a good fit <-> dBZ = 25 +1.5 dBR, nice for mapping order of magnitudes
… 25, 40, 55 dBZ -> 1, 10, 100 mm/h 
```

We've been looking into some of the methods and are trying to update references to have everything
up to date. Still a work in progress as I look through some of these old PRs and track down everything.

**Reply by raffyjef** [reply]
Thank you for this reply. I am asking this question because right now, I am also using a distrometer data and arrived at different Z-R relations (e.g. Z=339R1.19, Z=251R1.14, and etc) I am trying to find out on how to incorporate this to py-art. Is there a way to incorporate it? As I am trying to do so, I am using the mentioned pyart.retrieve.est_rain_rate_z function by changing the alpha and beta, however, it seems the alpha and beta in the function is not the same as the a and b in the Z=aR^b equation. 

Thanks for everything, hopefully you get what I am trying to point out.

**Reply by zssherman** [reply]
@raffyjef With the equation solving for rain rate, the coefficients in the function follow:
alpha = (1/A)^(1/b), beta = 1/b
or in your case:
 R = (1/339)^(1/1.19)Z^(1/1.19)

We are working on better documenting some of our older functions. I had to ask around to confirm as well.

**Reply by raffyjef** [reply]
I get it now. Thank you for this!

**Reply by zssherman** [reply]
Not a problem!

**Reply by raffyjef** [reply]
Sorry for this follow up question. Does the pyart.retrieve.est_rain_rate_kdp works the same for it's alpha and beta? I have a equation in the form R = a(kdp)^b. Thank you so much!

**Reply by zssherman** [reply]
It does not, in this case, the coefficients by default are being pulled from the kdp table for coefficients based on wavelengths for the S, C or X-sapr, so your coefficients should be fine as is without any added calculation prior

**Reply by raffyjef** [reply]
Thanks again!
